### ISSUE TITLE
Add sample softmax possibility to TransfoXL model for TransfoXL training

### ISSUE THREAD

**Issue by patrickvonplaten** [huggingface/transformers#3310]
# 🚀 Feature request

TransfoXL samples the logits during training if required. At the moment TransfoXL can only be used whithout sampling from the logits during training. A partrly finished implementation can be found under the branch: `add_sampling_and_training_to_transfo_xl_models` .

## Motivation

To be able to train tranfoXL correctly.

## Your contribution

Already looked into the issue. Could try to implement it correctly with help from @thomwolf and @LysandreJik . Not a priority at the moment though.
